src/SignalR/clients/csharp/Client.Core/src/HubConnectionExtensions.InvokeAsyncGeneric.cs

```diff
@@ -43,7 +43,7 @@ public static Task<TResult> InvokeAsync<TResult>(this HubConnection hubConnectio
         /// The <see cref="Task{TResult}.Result"/> property returns a <typeparamref name="TResult"/> for the hub method return value.
         /// </returns>
         [SuppressMessage("ApiDesign", "RS0026:Do not add multiple overloads with optional parameters", Justification = "Required to maintain compatibility")]
-        public static Task<TResult> InvokeAsync<TResult>(this HubConnection hubConnection, string methodName, object arg1, CancellationToken cancellationToken = default)
+        public static Task<TResult> InvokeAsync<TResult>(this HubConnection hubConnection, string methodName, object? arg1, CancellationToken cancellationToken = default)
```

Comment:
Should these be `Task<TResult?>`

Reply:
I don't think so? That would force callers to always handle null even if they never expect it to be null. If they use a nullable for `TResult` then that means they expect a null value and should handle it, and the compiler will let them know that. So seems better to let the user decide.

Reply:
Fair enough. It's effectively the same choice we made with JSInterop. But we also do not provide any guarantees against it which is crummy.